.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Static analysis
+run-name: GitHub Actions for clang-tidy
+
+on: pull_request
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: |
+          git remote add upstream "https://github.com/${{ github.event.pull_request.base.repo.full_name }}"
+          git fetch --no-tags --no-recurse-submodules upstream "${{ github.event.pull_request.base.ref }}"
+          git submodule update --init --recursive
+
+      - name: Install clang-tidy
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang clang-tidy libldap2-dev curl libcurl4-openssl-dev bison
```

Comment:
At https://github.com/VarunNagaraju/percona-server/actions/runs/7345240731/job/19998118252 I found:
```
CMake Warning at cmake/fido2.cmake:69 (MESSAGE):
  Cannot find development libraries.  You need to install the required
  packages:

    Debian/Ubuntu:              apt install libudev-dev
    RedHat/Fedora/Oracle Linux: yum install libudev-devel
    SuSE:                       zypper install libudev-devel
```
and
```
CMake Warning at cmake/kerberos.cmake:54 (MESSAGE):
  Cannot find KERBEROS and GSSAPI development libraries.  You need to install
  the required packages:

    Debian/Ubuntu:              apt install libkrb5-dev
    RedHat/Fedora/Oracle Linux: yum install krb5-devel
    SuSE:                       zypper install krb5-devel
```

Reply:
Will install them as well.